backend/src/transactions/user.go

```diff
@@ -16,3 +15,17 @@ func GetAllUsers(db *gorm.DB) ([]models.User, error) {
 
 	return users, nil
 }
+
+func UpdateUser(db *gorm.DB, id string, user models.User) (models.User, error) {
+	var existingUser models.User
+
+	if err := db.First(&existingUser, id).Error; err != nil {
+		return models.User{}, fiber.NewError(fiber.StatusNotFound, "User not found")
```

Comment:
what are the possible reasons for this error case? is it only that the user isn't found? take a peek at how this concept is handled [here](https://github.com/GenerateNU/sac/blob/main/backend/src/transactions/category.go#L12C6-L12C20)